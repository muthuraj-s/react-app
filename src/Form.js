
import React from 'react';
import Select from 'react-select';
import axios from 'axios';

const options = [
    { value: 'daily', label: 'daily' }, { value: 'monthly', label: 'monthly' }, { value: 'yearly', label: 'yearly' }
];
class Form extends React.Component {
    constructor() {
        super();
        this.state = {
            name: '',
            password: '',  
            data:[]       
        }
    }
   componentDidMount(){
        console.log("aaa");
       
            axios.get('http://localhost:3000/data/data.json').then((response) => {
                console.log(response);
                this.setState({ data: response.data })
            }).catch((error) => {
                console.log(error)
            })
        }
   
    handleChange = (e) => {
        const field = e.target.name;
        const value = e.target.value;
        this.setState({ [field] : value});
        
    }
    
    handleCheck = check => {
        this.setState({ check })
        // console.log(check)
    }
    
    handleSubmit=(e)=>{
        
        e.preventDefault();
    }
    handleDrop = selectedOption => {
        this.setState({ selectedOption } );
        // console.log(selectedOption)
    };

    render() { 
        return (
           


    <div className="row">
                  <div className="col-sm-2"><h4>Form</h4></div>
                <div style={{display:"flex",justifyContent:"center", alignItems:"center"}}>
               
                    <div style={{ height: '500px' }}>
                        <div style={{ textAlign: 'left', paddingTop: '20px' }}>
                            <label>Name </label><br />
                            <input type="text" className="form-control" style={{ height: '26px' }} value={this.state.name} onChange={this.handleChange} name="name" id="name" />
                        </div>
                        <div  style={{ textAlign: 'left', paddingTop: '10px' }}>
                            <label >Password </label><br />
                            <input type="password" className="form-control" style={{ height: '26px' }} value={this.state.password} onChange={this.handleChange} id="password" name="password" />
                        </div>

                        <div  style={{ textAlign: 'left', paddingTop: '10px' }}>
                            <label >Select preferences </label><br />
                            <input type="checkbox" onChange={this.handleCheck}  id="a"  />chennai
                                <input type="checkbox"  onChange={this.handleCheck} id="b"/>banglore
                                <input type="checkbox"  onChange={this.handleCheck} id="c"/>pune
                        </div>
                        <div  style={{ textAlign: 'left', paddingTop: '10px' }}>
                            <label >choose</label><br />
                            <Select
                                id="selectedOption"
                                onChange={this.handleDrop}
                                options={options}
                                name="selectedOption"
                            />
                            </div>
                      
                        <div  style={{ textAlign: 'left', paddingTop: '10px' }}>
                            <label >Terms&conditions</label><br />
                            <div>
                            <input name="group1"type="radio" id="a"  />agree&nbsp;&nbsp;
                                <input  name="group1" type="radio"  id="b"  />disagree
                               </div>
                        </div><br/>
                        <div className="row" align="center">
                    <button onClick={this.handleSubmit}><b>submit</b></button>
    
                    </div>
                </div>

               </div>
            </div>    
  
        )

    }
}
export default Form;