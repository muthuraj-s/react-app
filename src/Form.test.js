import React from 'react';
import { shallow, mount, render } from 'enzyme';

import Form from './Form';
import { shallowToJson } from 'enzyme-to-json';

describe('Rule Details Test Suite', () => {
     let component;
     let props;

    const shallowComponent = async () => {
        component = await shallow(<Form />);
        await component.instance().componentDidMount();
        component.update();
    };

    afterEach(() => {
        component = null;
        props = null;
        jest.clearAllMocks();
        sinon.restore();
    });
    describe('renders', () => {
        it('renders ', async () => {
         shallowComponent();
            expect(component).toBeDefined();
            expect(shallowToJson(component)).toMatchSnapshot();
        });
    });
});