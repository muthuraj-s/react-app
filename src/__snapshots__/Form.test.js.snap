// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rule Details Test Suite renders renders  1`] = `
<div
  className="row"
>
  <div
    className="col-sm-2"
  >
    <h4>
      Form
    </h4>
  </div>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <div
      style={
        Object {
          "height": "500px",
        }
      }
    >
      <div
        style={
          Object {
            "paddingTop": "20px",
            "textAlign": "left",
          }
        }
      >
        <label>
          Name 
        </label>
        <br />
        <input
          className="form-control"
          id="name"
          name="name"
          onChange={[Function]}
          style={
            Object {
              "height": "26px",
            }
          }
          type="text"
          value=""
        />
      </div>
      <div
        style={
          Object {
            "paddingTop": "10px",
            "textAlign": "left",
          }
        }
      >
        <label>
          Password 
        </label>
        <br />
        <input
          className="form-control"
          id="password"
          name="password"
          onChange={[Function]}
          style={
            Object {
              "height": "26px",
            }
          }
          type="password"
          value=""
        />
      </div>
      <div
        style={
          Object {
            "paddingTop": "10px",
            "textAlign": "left",
          }
        }
      >
        <label>
          Select preferences 
        </label>
        <br />
        <input
          id="a"
          onChange={[Function]}
          type="checkbox"
        />
        chennai
        <input
          id="b"
          onChange={[Function]}
          type="checkbox"
        />
        banglore
        <input
          id="c"
          onChange={[Function]}
          type="checkbox"
        />
        pune
      </div>
      <div
        style={
          Object {
            "paddingTop": "10px",
            "textAlign": "left",
          }
        }
      >
        <label>
          choose
        </label>
        <br />
        <StateManager
          defaultInputValue=""
          defaultMenuIsOpen={false}
          defaultValue={null}
          id="selectedOption"
          name="selectedOption"
          onChange={[Function]}
          options={
            Array [
              Object {
                "label": "daily",
                "value": "daily",
              },
              Object {
                "label": "monthly",
                "value": "monthly",
              },
              Object {
                "label": "yearly",
                "value": "yearly",
              },
            ]
          }
        />
      </div>
      <div
        style={
          Object {
            "paddingTop": "10px",
            "textAlign": "left",
          }
        }
      >
        <label>
          Terms&conditions
        </label>
        <br />
        <div>
          <input
            id="a"
            name="group1"
            type="radio"
          />
          agree  
          <input
            id="b"
            name="group1"
            type="radio"
          />
          disagree
        </div>
      </div>
      <br />
      <div
        align="center"
        className="row"
      >
        <button
          onClick={[Function]}
        >
          <b>
            submit
          </b>
        </button>
      </div>
    </div>
  </div>
</div>
`;
